From 935dc35c75318fa213d26808ad8bb130fb0b486e Mon Sep 17 00:00:00 2001
Message-Id: <935dc35c75318fa213d26808ad8bb130fb0b486e.1659673099.git.geliang.tang@suse.com>
From: Yixun Lan <dlan@gentoo.org>
Date: Wed, 6 Jul 2022 22:02:04 +0800
Subject: [PATCH] libbpf, riscv: Use a0 for RC register
Git-commit: 935dc35c75318fa213d26808ad8bb130fb0b486e
Patch-mainline: v6.0-rc1
References: bsn#210
Modified-by-SEL: Yes, refreshed due to context changes

According to the RISC-V calling convention register usage here [0], a0
is used as return value register, so rename it to make it consistent
with the spec.

  [0] section 18.2, table 18.2
      https://riscv.org/wp-content/uploads/2015/01/riscv-calling.pdf

Fixes: 589fed479ba1 ("riscv, libbpf: Add RISC-V (RV64) support to bpf_tracing.h")
Signed-off-by: Yixun Lan <dlan@gentoo.org>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Björn Töpel <bjorn@kernel.org>
Acked-by: Amjad OULED-AMEUR <ouledameur.amjad@gmail.com>
Link: https://lore.kernel.org/bpf/20220706140204.47926-1-dlan@gentoo.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/lib/bpf/bpf_tracing.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/lib/bpf/bpf_tracing.h
+++ b/tools/lib/bpf/bpf_tracing.h
@@ -207,7 +207,7 @@
 #define __PT_PARM5_REG a4
 #define __PT_RET_REG ra
 #define __PT_FP_REG s0
-#define __PT_RC_REG a5
+#define __PT_RC_REG a0
 #define __PT_SP_REG sp
 #define __PT_IP_REG pc
 
